Add vorpy package

jackericson98 opened the pull request:
This PR adds the VorPy package to conda-forge.
     
     VorPy is a Python package for Voronoi analysis of molecular structures.
     
     - Package: vorpy
     - Version: 3.0.1
     - License: MIT
     - Source: PyPI

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/vorpy/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/vorpy/meta.yaml**:

 * ❌ There are too few lines.  There should be one empty line at the end of the file.
 * ❌ license_file entry is missing, but is required.

For **recipes/vorpy/meta.yaml**:

 * ℹ️ Recipes should usually depend on `matplotlib-base` as opposed to `matplotlib` so that runtime environments do not require large packages like `qt`.
 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of the recipe, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).
 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15886842584](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15886842584). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/vorpy/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/vorpy/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15887051863](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15887051863). Examine the logs at this URL for more detail.</sub>

synapticarbors commented on recipes/vorpy/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+{% set name = "vorpy" %}
```
I think this should be `vorpy3` to match pypi:
https://pypi.org/project/vorpy3/

synapticarbors commented on recipes/vorpy/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Fixing the name will also fix the download issue

synapticarbors commented on recipes/vorpy/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+  imports:
+    - vorpy
+  commands:
+    - python -c "import vorpy; print(vorpy.__version__)"
```
This is redundant with the `imports` test. Please add a `pip check` though:

https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L64-L67

and also add `python {{ python_min }}` under `requires` as is shown in:
https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/vorpy/meta.yaml```, ```recipes/coarsify/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/vorpy/meta.yaml**:

 * ❌ There are too few lines.  There should be one empty line at the end of the file.

For **recipes/coarsify/meta.yaml**:

 * ℹ️ Recipes should usually depend on `matplotlib-base` as opposed to `matplotlib` so that runtime environments do not require large packages like `qt`.
 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.

   - For the `host` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of the recipe, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
- If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/31538897736](https://github.com/conda-forge/conda-forge-webservices/actions/runs/31538897736). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/vorpy/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/vorpy/meta.yaml**:

 * ℹ️ The recipe sets `python_min` to 3.9, which is equal or lower than the default provided by conda-forge's global pinning. Please remove the redefinition.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/32521538913](https://github.com/conda-forge/conda-forge-webservices/actions/runs/32521538913). Examine the logs at this URL for more detail.</sub>

jackericson98 commented:
Hi @synapticarbors, I’ve updated the recipe based on the requested changes and rebased/updated it against the current staged-recipes main branch. All CI checks are now passing on Windows, Linux, and macOS. When you have a chance, could you please take another look? Thanks!

eunos-1128 commented:
Probably, you should use the v1 recipe format before proceeding to be reviewed.

Thank you!

[docs.conda.io/projects/conda-build/en/latest/user-guide/v1-recipes.html#migrating-recipes](https://docs.conda.io/projects/conda-build/en/latest/user-guide/v1-recipes.html#migrating-recipes)

```
conda-recipe-manager convert recipe/meta.yaml > recipe/recipe.yaml
```

eunos-1128 commented:
@conda-forge/help-python, ready for review

ocefpaf commented:
Please convert to the latest, and faster, recipe format `v1`.

```
grayskull pypi --strict-conda-forge --use-v1-format <package-name>
```

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/vorpy/meta.yaml```, ```recipes/vorpy3/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipes/vorpy/meta.yaml**:

 * ℹ️ The recipe sets `python_min` to 3.9, which is equal or lower than the default provided by conda-forge's global pinning. Please remove the redefinition.

For **recipes/vorpy3/recipe.yaml**:

 * ❌ There are too few lines.  There should be one empty line at the end of the file.

For **recipes/vorpy3/recipe.yaml**:

 * ℹ️ `noarch: python` packages install on every Python version at or above `python_min`, but the Python test only runs against a single version. Consider testing against both the minimum and the latest supported Python:
```yaml
tests:
  - python:
      python_version:
        - ${{ python_min }}.*
        - "*"
```
 * ℹ️ The recipe sets `python_min` to 3.9, which is equal or lower than the default provided by conda-forge's global pinning. Please remove the redefinition.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/32748824515](https://github.com/conda-forge/conda-forge-webservices/actions/runs/32748824515). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/vorpy3/recipe.yaml```) and found it was in an excellent condition.
